File: HSE.RP.API/Models/EmploymentType.cs
using System;
using HSE.RP.API.Enums;

namespace HSE.RP.API.Models
{
    public record EmploymentType
    {
        public bool? PublicSector { get; set; }
        public bool? PrivateSector { get; set; }
        public bool? Other { get; set; }
        public bool? Unemployed { get; set; }

        public ComponentCompletionState CompletionState { get; set; } = ComponentCompletionState.NotStarted;
    }
}

File: HSE.RP.Client/src/app/models/employment-type.ts
import { ComponentCompletionState } from "./component-completion-state.enum";
import { IComponentModel } from "./component. interface";

export class EmploymentType implements IComponentModel {
    PublicSector?: boolean;
    PrivateSector?: boolean;
    Other?: boolean;
    Unemployed?: boolean;
    CompletionState?: ComponentCompletionState = ComponentCompletionState.NotStarted;
}
